scripts/utility.py

```diff
@@ -0,0 +1,49 @@
+import threading
+from functools import wraps
+from typing import Callable, Optional, TypeVar, overload
+
+from typing_extensions import ParamSpec
+
+try:
+    import thread
+except ImportError:
+    import _thread as thread
```

Comment:
When does the import of `thread` fail? I could only find the documentation for [`_thread`](https://docs.python.org/3/library/_thread.html#module-_thread) in the Python3 docs. Is `thread` just a Python2 thing, since it is available [here](https://python.readthedocs.io/en/v2.7.2/library/thread.html)? So just using `import _thread` would be enough.

Also weird that the module is protected, properly because it is a low-level API. But it should be fine since it's just for the publisher coverage and provides an easy way to interrupt the main thread.

Alternatively, we could avoid using `_thead` entirely by passing a mutable argument to the interpreter function. But I'm not sure if this is really more elegant. (Example [here](https://stackoverflow.com/a/25189669))

Reply:
Good point, I'm gonna change that import according to your comment.

I would avoid a solution using a mutable argument, but interesting idea anyways! Thanks for the insight.

Reply:
Yes, I agree with you.